lib/server/push.api.js

```diff
@@ -443,10 +443,16 @@ Push.Configure = function(options) {
 
     Meteor.setInterval(function() {
 
-        if (!isSendingNotification) {
-            // Set send fence
-            isSendingNotification = true;
+        if (isSendingNotification) {
+            return;
+        }
+        // Set send fence
+        isSendingNotification = true;
 
+        var countSent = 0;
+        var batchSize = options.sendBatchSize || 1;
+        while (Push.notifications.find().count() > 0 && countSent++ < batchSize) {
```

Comment:
@raix that definitely makes sense, I was thinking a single query per batch would be a good idea as well.  I’ll let you know when I have some time to work on it, unless someone else does it first :-)